orte/mca/notifier/syslog/configure.m4

```diff
@@ -0,0 +1,33 @@
+# -*- shell-script -*-
+#
+# Copyright (c) 2017 Cisco Systems, Inc.  All rights reserved.
+# $COPYRIGHT$
+#
+# Additional copyrights may follow
+#
+# $HEADER$
+#
+
+# MCA_notifier_syslog_CONFIG([action-if-found], [action-if-not-found])
+# -----------------------------------------------------------
+AC_DEFUN([MCA_orte_notifier_syslog_CONFIG], [
+    AC_CONFIG_FILES([orte/mca/notifier/syslog/Makefile])
+
+    OPAL_VAR_SCOPE_PUSH([orte_notifier_syslog_happy])
+
+    # Per https://github.com/open-mpi/ompi/issues/4373 and
+    # https://github.com/open-mpi/ompi/pull/4374, we need to check
+    # that syslog.h is compilable (we already checked this in the main
+    # configure.ac, but the result is cached / invisible to us -- so
```

Comment:
This comment is poorly phrased.  The result of the previous test *IS* available to the component.  The right way to get the result is to call AC_CHECK_HEADER() again.  Caching is a wonderful thing...

Reply:
K.  How about:

```sh
    # Per https://github.com/open-mpi/ompi/issues/4373 and
    # https://github.com/open-mpi/ompi/pull/4374, we need to check
    # that syslog.h is compilable (we already checked this in the main
    # configure.ac, but the result is not visible to us -- so check it
    # again and examine the result this time).  If syslog.h is not
    # compilable, disable this component.
```

Reply:
Why not just:

```shell
    # Per https://github.com/open-mpi/ompi/issues/4373 and
    # https://github.com/open-mpi/ompi/pull/4374, we need to check
    # that syslog.h is compilable.  If syslog.h is not
    # compilable, disable this component.
```

Calling AC_CHECK_HEADER and friends again and again is a common pattern in Open MPI's configure script.  It doesn't make sense to call that pattern out as somehow special in this one case.